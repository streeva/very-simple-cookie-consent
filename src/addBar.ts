import buildButton from './buildButton'
import buildWrapper from './buildWrapper'
import addGaScripts from './addGaScripts'

const hideBar = () => {
  document.querySelector('#sa-cc')?.classList.add('hiding')
  setTimeout(() => {
    document.querySelector('#sa-cc-wrapper')?.remove()
  }, 2000)
}

export default (cct: {
  cookieManager: { approve: () => void; decline: () => void }
}) => {
  const ccWrapper = buildWrapper()
  const btnsContainer = ccWrapper.querySelector('.sa-cc-bar__buttons')

  const approveHandler = (e: any) => {
    e.preventDefault()
    hideBar()
    console.log('Accept clicked')
    cct.cookieManager.approve()
    // load GA script
    addGaScripts()
  }

  const declineHandler = (e: any) => {
    e.preventDefault()
    hideBar()
    console.log('Decline clicked')
    cct.cookieManager.decline()
  }

  btnsContainer?.appendChild(
    buildButton(
      'cookie-notification__accept',
      'Accept',
      'sa-cc-bar__button--primary',
      ccWrapper,
      approveHandler
    )
  )

  btnsContainer?.appendChild(
    buildButton(
      'cookie-notification__decline',
      'Decline',
      'sa-cc-bar__button--secondary',
      ccWrapper,
      declineHandler
    )
  )

  document.body.appendChild(ccWrapper)
}
