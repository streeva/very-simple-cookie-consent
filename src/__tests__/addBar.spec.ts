import { expect, test, assert } from 'vitest'
import addBar from '../addBar'

test('addBar', () => {
  const policyLink = 'https://example.com/policy-link'
  assert.equal(document.body.toString(), '<body></body>', 'empty body')

  addBar(
    policyLink,
    () => {},
    () => {}
  )

  const policyLinkEle = document.querySelector('#vs-cc-bar__policy-link')
  expect(policyLinkEle?.getAttribute('href')).eq(policyLink)

  const buttons = document.querySelectorAll('button')
  expect(buttons.length).eq(2)

  expect(document.body.toString()).toMatchSnapshot()
})
