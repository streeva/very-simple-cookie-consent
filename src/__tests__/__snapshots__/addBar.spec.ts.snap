// Vitest Snapshot v1

exports[`addBar 1`] = `
"<body><div id=\\"vs-cc-wrapper\\">
<div id=\\"vs-cc\\" class=\\"vs-cc-bar\\">
  <div class=\\"vs-cc-bar__container\\">
    <div class=\\"vs-cc-bar__notification container\\" role=\\"alert\\">
      <div class=\\"vs-cc-bar__text\\">
        <h2 class=\\"vs-cc-bar__title\\">🍪 Can we use optional cookies?</h2>
        <p class=\\"no-margin\\">
          These cookies help us keep our website safe and give you a better
          experience. We won’t turn them on unless you accept. Want to know more
          or adjust your preferences? Here’s our
          <a id=\\"vs-cc-bar__policy-link\\" href=\\"https://example.com/policy-link\\">cookie policy</a>.
        </p>
      </div>
      <div class=\\"vs-cc-bar__actions\\">
        <div class=\\"vs-cc-bar__buttons\\"><button class=\\"vs-cc-bar__button vs-cc-bar__button--primary\\">Accept</button><button class=\\"vs-cc-bar__button vs-cc-bar__button--secondary\\">Decline</button></div>
      </div>
    </div>
  </div>
</div>
</div></body>"
`;
